backends/tfhe-cuda-backend/cuda/src/integer/multiplication.cuh

```diff
@@ -389,13 +389,16 @@ __host__ void host_integer_partial_sum_ciphertexts_vec(
                                       needs_processing);
 
     auto active_streams = streams.active_gpu_subset(total_ciphertexts);
+    GPU_ASSERT(total_ciphertexts <= mem_ptr->luts_message_carry->num_blocks,
```

Comment:
Should be a PANIC_IF_FALSE in my opinion, GPU_ASSERT will only trigger in debug mode